lib/Doctrine/Common/Cache/CacheProvider.php

```diff
@@ -239,6 +252,28 @@ protected function doFetchMultiple(array $keys)
     abstract protected function doContains($id);
 
     /**
+     * Default implementation of doSaveMultiple. Each driver that supports multi-put should overwrite it.
+     *
+     * @param array $keysAndValues  Array of keys and values to save in cache
+     * @param int   $lifetime       The lifetime. If != 0, sets a specific lifetime for these
+     *                              cache entries (0 => infinite lifeTime).
+     *
+     * @return bool TRUE if the operation was successful, FALSE if it wasn't.
+     */
+    protected function doSaveMultiple(array $keysAndValues, $lifetime = 0)
+    {
+        $success = true;
+
+        foreach ($keysAndValues as $key => $value) {
+            if (!$this->doSave($key, $value, $lifetime)) {
+                $success = false;
```

Comment:
This logic assumes that the cache is not going to bail out early when saving an item fails: this requires a test case with a failure.

I suggest coding this sort of test case against a mocked version of the `CacheProvider` that implements `doSave` that triggers a failure